### ISSUE TITLE
Connection closed after succesfull getting information

### ISSUE THREAD

**Issue by piotr313** [tananaev/traccar#1120]
Hello,

I'm trying to test Globalsat protocol and it seems that server closes connection right after getting correct report and sending ACK message to the device.
Can you tell me is it how it should work? Can you please point me where in the sources the connection is being handled ?

**Comment by tananaev** [comment]
Server shouldn't close connection. Can I see log file?

**Comment by piotr313** [comment]
2015-03-17 10:20:37  INFO: Starting server...
2015-03-17 10:20:37  INFO: Operating System name: Windows 7 version: 6.1 architecture: amd64
2015-03-17 10:20:37  INFO: Java Runtime name: Java HotSpot(TM) 64-Bit Server VM vendor: Oracle Corporation version: 25.40-b25
2015-03-17 10:20:37  INFO: Memory Limit heap: 1820mb non-heap: 0mb
2015-03-17 10:20:37  INFO: Version: 2.11-SNAPSHOT
2015-03-17 10:21:14 DEBUG: [1DEF43D8: 50434 <- 127.0.0.1] - HEX: 4753722c3335373933383032303331303731302c2c342c30342c2c312c3137303331352c3036303635372c4530303030302e303030302c4e303030302e303030302c3134382c302e30302c302c302c302e302c31313939316d562a3663210d0a
2015-03-17 10:21:14 DEBUG: [1DEF43D8: 50434 -> 127.0.0.1] - HEX: 416664434b0d
2015-03-17 10:21:14  WARN: For input string: "11991mV" - java.lang.NumberFormatException (null:-1)
2015-03-17 10:21:14  INFO: Closing connection by exception
2015-03-17 10:21:14  INFO: Closing connection by disconnect

**Comment by piotr313** [comment]
ok now i see where is the problem. I think it shouldn't send ACK command if there is decode probkem...

**Comment by tananaev** [comment]
You need to specify format of the messages in the config file if it's different from default:
<pre>format0 = "TSPRXAB27GHKLMnaicz*U!";
format1 = "SARY*U!";</pre>

**Comment by piotr313** [comment]
I set the format exactly the same. The problem is 'mv' in string with battery level

**Comment by piotr313** [comment]
Just to remove "mv" problem (GlobalsatProtocolDecoder):

                case 'N':
                     value =  value.replaceAll("[^0-9.]", "");
                    extendedInfo.set("battery", Double.valueOf(value));
                    break;

**Comment by tananaev** [comment]
If the format is the same how do you get `N` in the message? Can you please provide format string that you used?

**Comment by piotr313** [comment]
<entry key='globalsat.format0'>SORPZAB27GHKLMN*U!</entry> 
I took it from some other post on Traccar. Later noticed that not all the parameters are described in TR-600 manual, and some of them return nothing. This also was with 'N' - it didn't returned anything but 'n' returns - battery power with 'mv' at the end. So I have just changed the setting in the device (changed N to n) without changing the traccar setting. Probably should just add another case to the GlobalsatProtocolDecoder... Here is part of the manual.

n - Capacity of battery that
presently supplied to
TR-600
if main battery is connected, n=voltage of
main battery (eg, 12300=12.3V)
If main battery is not connected, TR-600
would use backup batteries, that is
lead-acid battery or Li-battery.
if voltage of lead-acid backup battery >=
5V, n=voltage of lead-acid backup battery
(eg, 11800=11.8V)
If voltage of lead-acid battery is lower
than 5V or lead-acid battery is not
connected, TR-600 would be
power-supplied by Li-battery. n=capacity
of Li backup battery (eg, 80%)
  
As I understand it may return either voltage or percentage... My returns Voltage as I have main power supply only.
